azure/src/wizard/ResourceGroupVerifyStep.ts

```diff
@@ -0,0 +1,84 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { ResourceManagementClient } from '@azure/arm-resources';
+import { ActivityChildItem, ActivityChildType, activityFailContext, activityFailIcon, AzureWizardExecuteStepWithActivityOutput, createContextValue, ExecuteActivityOutput, nonNullProp, parseError } from '@microsoft/vscode-azext-utils';
+import { v4 as uuidv4 } from "uuid";
+import { l10n, Progress, TreeItemCollapsibleState } from 'vscode';
+import * as types from '../../index';
+import { createResourcesClient } from '../clients';
+import { ext } from '../extensionVariables';
+
+// See for more background: https://github.com/microsoft/vscode-azuretools/pull/1992#issue-3034841865
+export class ResourceGroupVerifyStep<T extends types.IResourceGroupWizardContext> extends AzureWizardExecuteStepWithActivityOutput<T> {
+    public priority: number = 95;
+    public stepName: string = 'resourceGroupVerifyStep';
+
+    protected getOutputLogFail = (context: T) => l10n.t('Failed to verify whether resource group with name "{0}" already exists.', nonNullProp(context, 'newResourceGroupName'));
+    protected getOutputLogSuccess(context: T) {
+        return context.resourceGroup?.name ?
+            l10n.t('Verified resource group with name "{0}" already exists.  Using existing resource group.', context.resourceGroup.name) :
+            l10n.t('Verified resource group with name "{0}" is available for creation.', nonNullProp(context, 'newResourceGroupName'));
+    }
+    protected getTreeItemLabel(context: T) {
+        return context.resourceGroup?.name ?
+            l10n.t('Verify resource group "{0}" (exists)', context.resourceGroup.name) :
+            l10n.t('Verify resource group "{0}" (available)', nonNullProp(context, 'newResourceGroupName'));
+    }
+
+    public async execute(context: T, progress: Progress<{ message?: string; increment?: number }>): Promise<void> {
+        progress.report({ message: l10n.t('Checking resource group availability...') });
+
+        const newName: string = nonNullProp(context, 'newResourceGroupName');
```

Comment:
You don't need to check for `context.newResourceGroupName` || `context.resourceGroup?.name`? I thought that `newResourceGroupName` should be undefined if the user selected an existing resource group.

Reply:
Well it shouldn't execute if `context.resourceGroup` exists.  The original goal of this implementation was to extract the logic out of `ResourceGroupCreateStep` which also only executes when `context.resourceGroup` doesn't exist.  Do we want to expand the way that this step gets used?

Reply:
If that were the case, then I'm a little confused because there seems to be two scenarios: 1. when the resource group exists and 2. when the resource group is available.

Would 1. be when you are using an existing resource group and 2. be when you have created one?

Reply:
Yeah, in my opinion this is generally what should be handled on the `ResourceGroupNameStep` since we added the async validation method which allows us to check whether or not the name is a duplicate of an existing resource.

In this implementation I'm mostly just trying to preserve the existing behavior because I don't know all the design reasons that went into adding the verify logic in the execute phase.  

For example, I know we also sometimes generate related names automatically.  What I don't know is whether or not this logic helps support that in the sense that could the generated name accidentally be of an existing resource group

Reply:
> Would 1. be when you are using an existing resource group and 2. be when you have created one?

1 is when your `newResourceGroupName` is the duplicate of an existing resource, 2 is when you have a `newResourceGroupName` that isn't a duplicate